File: leetcode/typescript/75 problems set/gcd_of_string.ts

function gcd(num1, num2){
    while (num2!=0){
        [num1, num2] = [num2, num1%num2]
    }
    return num1
}

function gcdOfStrings(str1: string, str2: string): string {
    var leng;
    if (str1+str2!==str2+str1){return ""}
    leng = gcd(str1.length, str2.length);
    return str1.slice(0,leng);
};

File: leetcode/python/75 problems set/gcd_of_string.py
class Solution:
    def gcd(self, num1, num2):
        while num2!=0:
            num1,num2 = num2, num1%num2
        return num1

    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1+str2!=str2+str1:
            return ""
        lenght = self.gcd(len(str1), len(str2))
        return str1[0:lenght]